pysph/tools/ipython.py

```diff
@@ -0,0 +1,641 @@
+class Viewer(object):
+
+    '''A class for the pysph jupyter notebook Viewer2D.
+
+    Attributes
+    ----------
+
+        path              String containing the path of the output directory
+        paths_list        List of paths to the output files
+        cache             Cached data
+
+    Methods
+    -------
+
+        get_buffer
+        show_log
+        show_info
+        show_results
+        show_all
+
+    '''
+
+    def __init__(self, path, cache=True):
+
+        # Imports #
+
+        from pysph.solver.utils import get_files, load
+
+        # Configuring the path #
+
+        self.path = path
+        self.paths_list = get_files(path)
+
+        # Caching #
+        # Note : Caching is only used by get_buffer and widget handlers.
+        if cache:
+            self.cache = {}
+        else:
+            pass
+
+    def get_buffer(self, frame):
+        '''Return particle arrays for a given frame number with caching.
+
+
+        Parameters
+        ----------
+
+        frame : int
+
+        Returns
+        -------
+
+        A dictionary.
+
+        Examples
+        --------
+
+        >>> sample = Viewer2D('/home/deep/pysph/trivial_inlet_outlet_output/')
+        >>> sample.get_buffer(12)
+        {'arrays': {'fluid': <pysph.base.particle_array.ParticleArray at 0x7f3f7d144d60>,
+        'inlet': <pysph.base.particle_array.ParticleArray at 0x7f3f7d144b98>,
+        'outlet': <pysph.base.particle_array.ParticleArray at 0x7f3f7d144c30>},
+        'solver_data': {'count': 240, 'dt': 0.01, 't': 2.399999999999993}}
+
+
+        '''
+        if hasattr(self, "_cbars"):
+            if frame in self.cache.keys():
+                temp_data = self.cache[frame]
+            else:
+                from pysph.solver.utils import load
```

Comment:
Why can't you put the import at the top as you seem to need it anyway, it isn't optional so might as well put it at the top.